### ISSUE TITLE
Consider removing query from DatabaseException.toString()?

### ISSUE THREAD

**Issue by vesse** [EvidentSolutions/dalesbred#24]
[`DatabaseException.toString()`](https://github.com/EvidentSolutions/dalesbred/blob/c7b40ec0abea13aaf2805972f239fcd239828bc0/dalesbred/src/main/java/org/dalesbred/DatabaseException.java#L63) adds the query to the output if it has one. While this is nice when developing it is something that I would not like to see in our production logs where the queries are large and exceptions are wrapped. `Throwable` uses `cause.toString()` to set the message of a wrapped exception and thus both `toString()` and `getMessage()` for the exception I catch output the full query.

**Comment by komu** [comment]
I'm not sure about this. I'd want as much information as possible when diagnosing problems on production. (Note that the logged query contains also used parameters, so it's not like the same thing is logged every time.)

I might add a static flag to `DatabaseException`, allowing you to say something akin to `DatabaseException.setIncludeQueriesInExceptionMessages(false)`, but I'll have to think about that.

**Comment by vesse** [comment]
Any solution that removes the query would suit me. My use case, in addition to the large query, is one where legislation does not allow the query to be logged due to the parameters, so we now catch the exception, traverse up the chain and get the cause from `DatabaseException`, rewrap it, and then rethrow to be able to log only what is allowed.

**Comment by komu** [comment]
Right, I can see your pain. I'll implement a solution for this.

**Comment by komu** [comment]
By the way, back in the day before I released 1.0, Dalesbred used to have [SqlQuery.confidential](https://github.com/EvidentSolutions/dalesbred/blob/v0.8.0/dalesbred/src/main/java/fi/evident/dalesbred/SqlQuery.java#L173) that one could use to wrap values into confidential boxes that produce `****` as their `toString`-output, but otherwise are bound normally to database. So you could say something like this:

``` java
db.update("insert into person (ssn, name) values (?, ?)", confidential(ssn), name);
```

I dropped it from 1.0 because it gave a false sense of security: even though Dalesbred itself would never print up the value, it might show up in the exception produced by JDBC-driver.

The same thing applies here:

``` java
try {
    db.update("CREATE TABLE person (id SERIAL PRIMARY KEY, ssn VARCHAR(11) NOT NULL UNIQUE, name VARCHAR NOT NULL)");

    db.update("INSERT INTO person (ssn, name) VALUES (?, ?)", "123456-0123", "Alice");
    db.update("INSERT INTO person (ssn, name) VALUES (?, ?)", "123456-0123", "Bob");
} catch (DatabaseSQLException e) {
    System.out.println(e.getCause());
}
```

Running this against PostgreSQL produces the following output:

```
org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "person_ssn_key"
  Detail: Key (ssn)=(123456-0123) already exists.
```

Oops...

Moral of the story: if you have strict requirements on what you are allowed to log, don't log messages of arbitrary exceptions.

So again, I'm on the fence here. I'd like to avoid implementing features that give a false sense of security on things that are inherently insecure. I'm all ears for good ideas, though.
